src/v/cluster/topics_frontend.cc

```diff
@@ -298,6 +301,58 @@ topics_frontend::create_non_replicable_topics(
       });
 }
 
+ss::future<std::vector<topic_result>>
+topics_frontend::dispatch_create_non_replicable_to_leader(
+  model::node_id leader,
+  std::vector<non_replicable_topic> topics,
+  model::timeout_clock::duration timeout) {
+    vlog(clusterlog.trace, "Dispatching create topics to {}", leader);
+    return _connections.local()
+      .with_node_client<cluster::controller_client_protocol>(
+        _self,
+        ss::this_shard_id(),
+        leader,
+        timeout,
+        [topics, timeout](controller_client_protocol cp) mutable {
+            return cp.create_non_replicable_topics(
+              create_non_replicable_topics_request{std::move(topics), timeout},
+              rpc::client_opts(model::timeout_clock::now() + timeout));
+        })
+      .then(&rpc::get_ctx_data<create_topics_reply>)
+      .then(
+        [topics = std::move(topics)](result<create_topics_reply> r) mutable {
+            if (r.has_error()) {
+                return create_non_replicable_topic_results(
+                  topics, map_errc(r.error()));
+            }
+            return std::move(r.value().results);
+        });
+}
+
+ss::future<std::vector<topic_result>>
+topics_frontend::create_non_replicable_topics(
```

Comment:
This code is nearly identical to `autocreate_topics`. I'm thinking,
1. Break out into separate common method that autocreate topics can also invoke
2. Use the `create_topics` API (probably more work)